evalml/pipelines/component_graph.py

```diff
@@ -96,8 +97,17 @@ def fit_features(self, X, y):
             X (pd.DataFrame): The input training data of shape [n_samples, n_features]
             y (pd.Series): The target training data of length [n_samples]
         """
-        self._compute_features(self.compute_order[:-1], X, y, fit=True)
-        return self
+        if len(self.compute_order) <= 1:
+            return X
+
+        component_outputs = self._compute_features(self.compute_order[:-1], X, y=y, fit=True)
+        final_component_inputs = []
```

Comment:
Just curious, what prompted this change?

Reply:
`Pipeline._compute_features_during_fit`, which calls this method, needs an `X_t` to pass to the final estimator